app/scripts/lib/nonce-tracker.js

```diff
@@ -12,15 +12,14 @@ class NonceTracker {
   // releaseLock must be called
   // releaseLock must be called after adding signed tx to pending transactions (or discarding)
   async getNonceLock (address) {
+    const pendingTransactions = this.getPendingTransactions(address)
     // await lock free
-    await this.lockMap[address]
+    if (pendingTransactions.length) await this.lockMap[address]
```

Comment:
we agreed to remove the pendingTransactions check before awaiting the nonce lock
(but you didnt remember why you put this here)
basically this (line 17-18) should look as it did before, simply `await this.lockMap[address]`